src/DotQuery.Core/AsyncQueryExecutor.cs

```diff
@@ -42,6 +42,23 @@ public Task<TResult> QueryAsync(TQuery query)
         }
 
         /// <summary>
+        /// Execute the given query as an asynchronous operation with a specified cache policy.
+        /// </summary>
+        /// <param name="query">The query to be executed</param>
+        /// <param name="options">The cache policy options.</param>
+        /// <returns>The task object representing the asynchronous operation</returns>
+        public Task<TResult> QueryAsync(TQuery query, CacheEntryOptions options)
```

Comment:
Any reason we need a standalone `CacheEntryOptions`? Would be great if we integrate cache entry option into exsiting QueryOptions class (as a field like CacheOption).